Backend MVP

tomquisel opened the pull request:
This is a viable MVP!

Key changes:

- Supports SQL scraper
- adds config file at `.env`
- stores more fields from clinicaltrials.gov
- renames institutions to organizations to be consistent with clinicaltrials.gov naming
- switches from computing derived fields at read time to computing at write time (this fixes performance issues with 100k+ trials are loaded)
- computes aggregate fields (like count of on-time trials) at write time using triggers
- makes setup more configurable, so both dev and prod environments can be run from same `docker-compose.yml`

tomquisel committed:
Support Scraper that writes direct to DB (#1)

* Make DB schema & config support the sql scraper

* Make restart more configurable

* Add a default development config

* Improve configurability

* Dial back the scraper limit for development

tomquisel committed:
Add new counts and make them sortable

tomquisel committed:
Add new fields, keep same field names as clinicaltrials.gov

tomquisel committed:
Improve database performance, features

- Add pre-computed columns to trials table
- Add trigger to pre-compute organization aggregate columns
- add new aggregate columns (fractions) to support frontend

tomquisel committed:
Make ports fully configurable

tomquisel committed:
Fix postgres port setting

tomquisel committed:
Fix misnamed variable

tomquisel committed:
Fix aggregation trigger

dstuck commented on db/init/00-database.sql:
```diff
@@ -1,30 +1,59 @@
+  late_count BIGINT DEFAULT 0,
+  missing_count BIGINT DEFAULT 0,
+  on_time_count BIGINT DEFAULT 0,
+  late_frac FLOAT DEFAULT 0.5,
```
Is there a reason this shouldn't be 0 or None?

tomquisel replied:
Yeah, that's better. Switched it to `NULL`

dstuck reviewed (approved):
Awesome!

tomquisel committed:
Address feedback: null fractions when 0 denominator

Also, switch to `INT` from `BIGINT` so the field can be represented as
a number in javascript
